cdap-docs/build.sh

```diff
@@ -489,36 +496,112 @@ function build_license_dependency_pdfs() {
 }
 
 function build_standalone() {
-  local add_artifacts=""
+  local errors
+  local check="0"
+  local additional_artifacts_path="/tmp/${ADDITIONAL_ARTIFACTS}"
+  local add_artifacts="-Dadditional.artifacts.dir=${additional_artifacts_path}"
+  rm -rf ${additional_artifacts_path}
+  mkdir -p ${additional_artifacts_path}
+  echo "Cleaned directory ${additional_artifacts_path}"
   if [ -d ${HYDRATOR_PLUGINS_PATH} ]; then
-    build_hydrator_plugins
-    local errors=$?
-    if [ "${errors}" == "0" ]; then
-      add_artifacts="-Dadditional.artifacts.dir=${HYDRATOR_PLUGINS_PATH}"
+    build_hydrator_plugins ${additional_artifacts_path}
+    errors=$?
+    if [ "${errors}" != "0" ]; then
+      echo "Error building Hydrator plugins: ${errors}"
+      check=1
     fi
   else
     echo "No HYDRATOR_PLUGINS_PATH at ${HYDRATOR_PLUGINS_PATH}"
+    check=1
+  fi
+  check_if_continue ${check}
+  check="0"
+  local add_apps=""
+  if [ -d ${TRACKER_APP_PATH} ]; then
+    build_tracker_app ${additional_artifacts_path}
+    errors=$?
+    if [ "${errors}" != "0" ]; then
+      echo "Error building Tracker app: ${errors}"
+      check=1
+    fi
+  else
+    echo "No TRACKER_APP_PATH at ${TRACKER_APP_PATH}"
+    check=1
```

Comment:
It's best to change all of this into a more generic system. Allow something like specifying multiple paths in a variable and iterate over it. EXTERNAL_PROJECT_PATHS="/some/path/to/hydrator-plugins /another/path/to/tracker /some/new/repo/not/created/yet"... you can symlink these locations into `${additional_artifacts_path}` and that should be all that you need... this will give you the support that you want, future-proof the code, and keep code specifically for private repositories outside public repos.